flake8_trio/visitors/flake8triovisitor.py

```diff
@@ -23,7 +23,9 @@
 
 class Flake8TrioVisitor(ast.NodeVisitor, ABC):
     # abstract attribute by not providing a value
-    error_codes: dict[str, str]  # pyright: ignore[reportUninitializedInstanceVariable]
+    error_codes: ClassVar[
+        dict[str, str]
+    ]  # pyright: ignore[reportUninitializedInstanceVariable]
```

Comment:
It is not! And turns out `reportUnnecessaryTypeIgnoreComment` doesn't default to true with `strict = true`, so time to turn that on.
I did change this one hoping it would flow through and silence RUF012 ("Mutable class attribute should be annotated with `typing.ClassVar`") to the classes inheriting from it - but when that didn't happen I silenced the error rather than change it in the 25 other places. But maybe I'll bother doing the big search&replace

Reply:
oh hm, the reason it wasn't enabled is because there's no separate setting for `pyright: ignore` vs `type: ignore` ([and unlikely to be added](https://github.com/python/mypy/issues/12358)), and [mypy doesn't have `mypy: ignore`](https://github.com/python/mypy/issues/12358)

Reply:
Could perhaps enable mypy's `type: ignore`, but that'd involve adding a bunch of deps to its pre-commit env and/or go through a bunch of them and replace with `pyright: ignore` - and I don't think that's worth bothering with currently.

Reply:
If we annotate as `Mapping[str, str]` so that it's logically immutable maybe our typecheckers will be happier?